ray-operator/test/e2erayservice/rayservice_upgrade_test.go

```diff
@@ -54,11 +54,10 @@ func TestOldHeadPodFailDuringUpgrade(t *testing.T) {
 
 	svcName := utils.GenerateServeServiceName(rayService.Name)
 	LogWithTimestamp(test.T(), "Checking that the K8s serve service %s has exactly one endpoint because the cluster only has a head Pod", svcName)
-	endpoints, err := test.Client().Core().CoreV1().Endpoints(namespace.Name).Get(test.Ctx(), svcName, metav1.GetOptions{})
+	readyEndpoints, err := GetReadyEndpointsFromSlices(test.Ctx(), test.Client(), namespace.Name, svcName)
 	g.Expect(err).NotTo(HaveOccurred())
-	g.Expect(endpoints.Subsets).To(HaveLen(1))
-	g.Expect(endpoints.Subsets[0].Addresses).To(HaveLen(1))
-	headPodName := endpoints.Subsets[0].Addresses[0].TargetRef.Name
+	g.Expect(readyEndpoints).To(HaveLen(1))
+	headPodName := readyEndpoints[0].TargetRefName
```

Comment:
Should we use also add  
```suggestion
	g.Expect(readyEndpoints).To(HaveLen(1))
	headPodName := readyEndpoints[0].TargetRefName
	headPodUID := readyEndpoints[0].TargetRefUID
```